viewer/packages/tools/src/Toolbar/ToolbarTool.ts

```diff
@@ -0,0 +1,181 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import { Cognite3DViewer } from '@reveal/core';
+import { DefaultCameraManager } from '@reveal/camera-manager';
+import { Cognite3DViewerToolBase } from '../Cognite3DViewerToolBase';
+import { Toolbar, ToolbarPosition } from './Toolbar';
+import { AxisViewTool } from '../AxisView/AxisViewTool';
+
+import svgAxisIcon from './icons/compass.svg';
+import svgScreenshotIcon from './icons/camera.svg';
+import svgCameraTargetIcon from './icons/zoomtotarget.svg';
+import svgZoomPastIcon from './icons/zoompastobject.svg';
+import svgFitToCameraIcon from './icons/fittocamera.svg';
+
+/**
+ * Tool to help user to use the default toolbar items such as Camera & other features in Reveal to enable/disable it.
+ * The tool also helps in adding or create custom toolbar.
+ */
+export class ToolbarTool extends Cognite3DViewerToolBase {
+  private readonly _viewer: Cognite3DViewer;
+  private readonly _toolbar: Toolbar;
+  private axisTool: AxisViewTool | undefined;
+  private readonly _cameraOptions: boolean[] = [false, false];
```

Comment:
There should not be any reason to have this state at all - it can be determined from the camera manger. Storing state should almost always be avoided when possible.